File: Exercism Codes/Python/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if len(hey_bob) == 0:
        return "Fine. Be that way!"
    if hey_bob[-1] == '?' and hey_bob == hey_bob.upper() and any(x.isalpha() for x in hey_bob):
        return "Calm down, I know what I'm doing!"
    if hey_bob[-1] == '?':
        return "Sure."
    if hey_bob == hey_bob.upper() and any(x.isalpha() for x in hey_bob):
        return "Whoa, chill out!"

    return "Whatever."

File: Exercism Codes/C#/Bob.cs
using System;
using System.Text.RegularExpressions;
public static class Bob
{
    public static string Response(string statement)
    {
        bool endsWithQuestionMark = statement.Trim().EndsWith("?");
        bool isUppercase = statement.Equals(statement.ToUpper()) && Regex.IsMatch(statement, @"[a-zA-Z]");
        if (string.IsNullOrWhiteSpace(statement))
        {
            return "Fine. Be that way!";
        }
        else if (endsWithQuestionMark && isUppercase)
        {
            return "Calm down, I know what I'm doing!";
        }
        else if (isUppercase)
        {
            return "Whoa, chill out!";
        }
        else if (endsWithQuestionMark)
        {
            return "Sure.";
        }
        else
        {
            return "Whatever.";
        }
    }
}